### DISCUSSION TITLE
Server-side rendering of custom content to React components

### DISCUSSION THREAD

**Discussion by Svish** [vercel/next.js#23898] [Help]
I'm getting some XML from an API that I have to convert into React-components and HTML. I've used `xmldom` to parse the XML-string and a small helper-class to walk through the XML-nodes, converting them into React- or HTML-elements, depending on what node it is. Works great.

But, just noticed that the data being sent to the client is the XML-string, and the `xmldom` dependency ends up in the client-side bundle.

Is there a way I can avoid having `xmldom` end up as part of the client-side dependencies?

The pages are SSG, using `getStaticProps`, and I'd like them to be as "thin" as possible.

**Comment by MrFussyfont** [comment]
I've noticed something similar with my site. The content is stored in Markdown and JSON files, rendered to HTML, but with SSG those source files are included unnecessarily (appended to the HTML), doubling the size.
